rllib/models/tf/complex_input_net.py

```diff
@@ -64,6 +64,7 @@ def __init__(self, obs_space, action_space, num_outputs, model_config, name):
                     else get_filter_config(obs_space.shape),
                     "conv_activation": model_config.get("conv_activation"),
                     "post_fcnet_hiddens": [],
+                    "vf_share_layers": True,
```

Comment:
is there a reason to make this the default value, with respect to this pr?

Reply:
Conceptually, I think it doesn't make sense to use separate vf layers in the CNN feature extractor, since complex input net doesn't actually use the value output of the CNN. So have separate vf layers just creates extra tensorflow layers that aren't needed.
On a practical level, this PR only fixes the shared vf layers case (including in visionnet itself), so keeping this at False doesn't work.

Reply:
can we use model_config.get("vf_share_layers", False) here, so that you are unblocked, but we don't change the default behavior?
we can even print a warming message here saying True would not actually work?

Reply:
Yes, we could do that, and that should unblock the PR. But just to double check, does ComplexInputNet even support separate vf layers? Unless I am missing something, it doesn't even look at the `vf_share_layers` config parameter. I think the network it builds is always shared up until the output layers.

Reply:
I thought this bit controls the structure of the vision net which does have different structures based on this bit?
are you saying vf_share_layers=False doesn't even work? should we just delete the False branch in visionnet.py then, and we don't have to mess with this bit here?

Reply:
Yes, that's right, this controls the structure of the visionnet. The False branch in visionnet works if the final layer output size is 1x1, but it doesn't work if it's any other size. So I don't think we'd want to delete the entire False branch in visionnet.py, as that might be used by someone.

I didn't get to the bottom of why exactly having this set to False fails here. Since ComplexInputNet doesn't even use the CNN's value output, it shouldn't really make a difference. I can check into that some more.

On the other hand, since ComplexInputNet never even uses the CNN's value output, I think it also really doesn't make a difference if we just always set this to True. 

My previous point was just that having `vf_share_layers` set to False in the overall model config passed into ComplexInputNet is a little strange, since ComplexInputNet just ignores that parameter and builds shared layers.What about if we keep it set to True for the CNN, and instead warn if ComplexInputNet gets it as False? Since that doesn't do what a user might expect it to do.